internal/pkg/vault/management.go

```diff
@@ -0,0 +1,193 @@
+			JSONObject:           &request,
+			BodyReader:           nil,
+			OperationDescription: "unseal secret store",
+			ExpectedStatusCode:   http.StatusOK,
+			ResponseObject:       &response,
+		})
+
+		if err != nil {
+			c.lc.Error(fmt.Sprintf("Error applying key share %d/%d: %s", keyCounter, secretShares, err.Error()))
+			return err
+		}
+
+		c.lc.Info(fmt.Sprintf("Vault key share %d/%d successfully applied.", keyCounter, secretShares))
+		if !response.Sealed {
+			c.lc.Info("Vault key share threshold reached. Unsealing complete.")
+			return nil
+		}
+		keyCounter++
+	}
+
+	return fmt.Errorf("%d", 1)
+}
+
+func (c *Client) InstallPolicy(token string, policyName string, policyDocument string) error {
+	_, err := c.doRequest(RequestArgs{
+		AuthToken:            token,
+		Method:               http.MethodPut,
+		Path:                 fmt.Sprintf(CreatePolicyPath, url.PathEscape(policyName)),
+		JSONObject:           UpdateACLPolicyRequest{Policy: policyDocument},
+		BodyReader:           nil,
+		OperationDescription: "install policy",
+		ExpectedStatusCode:   http.StatusNoContent,
+		ResponseObject:       nil,
+	})
+
+	return err
+}
+
+func (c *Client) EnableKVSecretEngine(token string, mountPoint string, kvVersion string) error {
+	urlPath := path.Join(MountsAPI, mountPoint)
+	parameters := EnableSecretsEngineRequest{
+		Type:        KeyValue,
+		Description: "key/value secret storage",
+		Options: &SecretsEngineOptions{
+			Version: kvVersion,
+		},
+	}
+
+	_, err := c.doRequest(RequestArgs{
+		AuthToken:            token,
+		Method:               http.MethodPost,
+		Path:                 urlPath,
+		JSONObject:           parameters,
+		BodyReader:           nil,
+		OperationDescription: "update mounts",
+		ExpectedStatusCode:   http.StatusNoContent,
+		ResponseObject:       nil,
+	})
+
+	return err
+}
+
+func (c *Client) EnableConsulSecretEngine(token string, mountPoint string, defaultLeaseTTL string) error {
+	urlPath := path.Join(MountsAPI, mountPoint)
+	parameters := EnableSecretsEngineRequest{
+		Type:        Consul,
+		Description: "consul secret storage",
+		Config: &SecretsEngineConfig{
+			DefaultLeaseTTLDuration: defaultLeaseTTL,
+		},
+	}
+
+	_, err := c.doRequest(RequestArgs{
+		AuthToken:            token,
+		Method:               http.MethodPost,
+		Path:                 urlPath,
+		JSONObject:           parameters,
+		BodyReader:           nil,
+		OperationDescription: "update mounts for Consul",
+		ExpectedStatusCode:   http.StatusNoContent,
+		ResponseObject:       nil,
+	})
+
+	return err
+}
+
+func (c *Client) CheckSecretEngineInstalled(token string, mountPoint string, engine string) (bool, error) {
+	var response ListSecretEnginesResponse
+
+	_, err := c.doRequest(RequestArgs{
+		AuthToken:            token,
+		Method:               http.MethodGet,
+		Path:                 MountsAPI,
+		JSONObject:           nil,
+		BodyReader:           nil,
+		OperationDescription: "query mounts for Consul",
```

Comment:
Please update the description based on the engine type:
```suggestion
		OperationDescription: "query mounts for "+engine,
```

Reply:
@jim-wang-intel , where are you getting `engine` from? This API is specific to Consul, but.. we could combine the two into `EnableSecretEngine` which is where I think you are going... ;-)

Reply:
Naw, that gets a little to messy... :-(

Reply:
Yup, good catch.